bigquery/cloud-client/snippets/src/main/java/com/example/bigquery/GrantAccessToDataset.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.bigquery;
+
+// [START bigquery_grant_access_to_dataset]
+import com.google.cloud.bigquery.Acl;
+import com.google.cloud.bigquery.Acl.Group;
+import com.google.cloud.bigquery.Acl.Role;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.Dataset;
+import com.google.cloud.bigquery.DatasetId;
+import java.util.ArrayList;
+import java.util.List;
+
+public class GrantAccessToDataset {
+
+  public static void main(String[] args) {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project and dataset from which to get the access policy
+    String projectId = "MY_PROJECT_ID";
+    String datasetName = "MY_DATASET_NAME";
+    // Create a new ACL granting the READER role to the group "user-or-group-to-add@example.com"
+    // For more information on the types of ACLs available see:
+    // https://cloud.google.com/storage/docs/access-control/lists
+    Acl newEntry = Acl.of(new Group("group-to-add@example.com"), Role.READER);
```

Comment:
question: should the initialization of `newEntry` be moved into the body of the sample? I think, for testing purposes, we will want to avoid passing around non-primitives for methods.